core/drand_daemon_control.go

```diff
@@ -138,13 +138,29 @@ func (dd *DrandDaemon) Shutdown(ctx context.Context, in *drand.ShutdownRequest)
 		bp.Stop(ctx)
 
 		dd.RemoveBeaconHandler(beaconID, bp)
-		dd.RemoveBeaconProcess(beaconID)
+		dd.RemoveBeaconProcess(beaconID, bp)
 	}
 
 	metadata := common.NewMetadata(dd.version.ToProto())
 	return &drand.ShutdownResponse{Metadata: metadata}, nil
 }
 
+// ReloadBeacon
+func (dd *DrandDaemon) ReloadBeacon(ctx context.Context, in *drand.ReloadBeaconRequest) (*drand.ReloadBeaconResponse, error) {
+	bp, beaconID, _ := dd.getBeaconProcess(in.GetMetadata())
```

Comment:
It seems weird to structure this such that the expected usage is for it to return an error.
Maybe we split it up so that there's a method to derive the beaconID separate from the call to get the processes if it exists?